drafts/inlinemacros.md

```diff
@@ -62,16 +65,19 @@ is equivalent to
 
 ## Inline Reductions
 
-The compiler will do the following rewritings.
+The compiler will do the following rewritings when encountering certain patterns of code
+outside the bodies of inline values and inline methods.
+When these patterns of code appear inside the aforementioned constructs, 
+they will be ignored by the inliner and processed by the compiler as usual.
 
-1. If `prefix.f[Ts](args1)...(argsN)` refers to a full applied inline
+1. If `prefix.f[Ts](args1)...(argsN)` refers to a fully applied inline
 method, where the prefix `prefix` and all arguments `argsI` for by-value
-parameters are side-effect free path expressions, replace the expression 
-with the method's right hand side, where parameter references are replaced 
+parameters are side-effect free path expressions or values of inline types, 
```

Comment:
> they are supposed to be side-effect free

If I got it right inline values are Function0's from semantic point of view, ie could have side effects.
If not, how could we check absence of side effects?